controllers/mysql_conf.go

```diff
@@ -0,0 +1,198 @@
+package controllers
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/cybozu-go/moco"
+	"github.com/go-logr/logr"
+)
+
+var (
+	confKeyPrefixes = []string{
+		"enable_",
+		"disable_",
+		"skip_",
+	}
+
+	// Default options of mysqld section
+	defaultMycnf = map[string]string{
+		"tmpdir":               moco.TmpPath,
+		"innodb_tmpdir":        moco.TmpPath,
+		"character_set_server": "utf8mb4",
+		"collation_server":     "utf8mb4_unicode_ci",
+
+		"default_time_zone": "+0:00",
+
+		"back_log":            "900",
+		"max_connections":     "100000",
+		"max_connect_errors":  "10",
+		"max_allowed_packet":  "1G",
+		"max_heap_table_size": "64M",
+		"sort_buffer_size":    "4M",
+		"join_buffer_size":    "2M",
+		"thread_cache_size":   "100",
+		"wait_timeout":        "604800", // == 7 days
+		"lock_wait_timeout":   "60",
+
+		// The default setting causes MY-012102. Adjust table_open_cache to suppress MY-012102,
+		// since　open_files_limit and table_open_cache are interdependent and those values are set dynamically
+		// See https://dev.mysql.com/doc/refman/8.0/en/server-error-reference.html#error_er_ib_msg_277
+		"table_open_cache":       "65536",
+		"table_definition_cache": "65536", // mitigate a innodb table cache eviction.
+
+		"transaction_isolation": "READ-COMMITTED",
+		"tmp_table_size":        "64M",
+		"slow_query_log":        "ON",
+		"long_query_time":       "2",
+		"log_error_verbosity":   "3",
+
+		"max_sp_recursion_depth": "20",
+
+		"print_identified_with_as_hex": "ON",
+
+		"loose_binlog_transaction_compression": "ON", // It would reduce the size of binlog by a third, but only available in 8.0.20 or later.
+
+		"information_schema_stats_expiry": "0", // No need to cache information_schema.tables values
+
+		"disabled_storage_engines": "MyISAM",
+
+		// INNODB Specific options
+		"innodb_flush_method":              "O_DIRECT",
+		"innodb_lock_wait_timeout":         "60",
+		"innodb_print_all_deadlocks":       "1",
+		"innodb_online_alter_log_max_size": "1073741824",
+		"innodb_adaptive_hash_index":       "ON",
+		"innodb_numa_interleave":           "ON",
+		"innodb_buffer_pool_in_core_file":  "OFF", // It is rarely necessary to include a buffer pool in a core file.
+
+		// options for adaptive flushing
```

Comment:
These are not only for the adaptive flushing.  Remove the comment and a blank line.

```suggestion
```

Reply:
committed.